users/views.py

```diff
@@ -1,3 +1,41 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.views           import View
+from django.http            import JsonResponse
+
+from users.models           import User
+from suulgo.settings        import SECRET_KEY, ALGORITHM
+
+class LoginInfoView(View):
+    def get(self, request):
+
+        user = request.user
+
+        user_info = User.objects.filter(id=user.id)
+
+        results = [
+            {
+                'name': user.name,
+                'profile_image_url': user.profile_image_url,
+                'email': user.email,
+                'survey_info': [
+                    {
+                        'class_number': survey.class_number,
+                        'stack': survey.stack,
+                        'alcohol_limit': survey.alcohol_limit,
+                        'alcohol_level': survey.alcohol_level,
+                        'comment': survey.comment,
+                        'favorite_place': survey.favorite_place,
+                        'favorite_food': survey.favorite_food,
+                        'hobby': survey.hobby,
+                        'gender': survey.gender.name,
+                        'mbti_name': survey.mbti.name,
+                        'mbti_information': survey.mbti.information
+                    }
+                    for survey in user.survey_set.all()
+                ]
+            }
+            for user in user_info
+        ]
```

Comment:
로그인 한 유저가 본인의 프로필 정보를 호출하는 view가 맞다면 suggestion과 같이 수정해야 할 것 같습니다.

- 데코레이터로부터 전달받은 request.user를 활용하려고 하셨지만 데코레이터가 작성되어 있지 않아 추가해줘야 합니다.
- 그러면 그 유저의 해당하는 정보만 불러오면 될텐데, 어차피 하나의 user 인스턴스만 반환되는 filter 메소드를 써서 for loop까지 이어지는 것이 불필요해 보입니다.
- 추가로 `=`과 `:`을 기준으로 정렬해주세요~

```suggestion
    @log_in_decorator
    def get(self, request):
        user    = request.user
        results = {
            'name': user.name,
            'profile_image_url': user.profile_image_url,
            'email': user.email,
            'survey_info': [{
                'class_number': survey.class_number,
                'stack': survey.stack,
                'alcohol_limit': survey.alcohol_limit,
                'alcohol_level': survey.alcohol_level,
                'comment': survey.comment,
                'favorite_place': survey.favorite_place,
                'favorite_food': survey.favorite_food,
                'hobby': survey.hobby,
                'gender': survey.gender.name,
                'mbti_name': survey.mbti.name,
                'mbti_information': survey.mbti.information
            } for survey in user.survey_set.all() ]
        }
```

Reply:
말씀대로 filter 메소드를 사용하는건 별의미가 없는거 같아요, 수정했습니다!